dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EntryDAO.java

```diff
@@ -242,6 +243,16 @@ private long countAllPublished() {
         return list(this.currentSession().getNamedQuery("io.dockstore.webservice.core.Entry.findLabelByEntryId").setParameter("entryId", entryId));
     }
 
+    public List<String> getToolsDescriptorTypes(long entryId) {
+        return (List<String>)this.currentSession().getNamedQuery("Entry.findToolsDescriptorTypes").setParameter("entryId", entryId)
+                .getSingleResult();
+    }
+
+    public List<String> getWorkflowsDescriptorTypes(long entryId) {
+        List result = this.currentSession().getNamedQuery("Entry.findWorkflowsDescriptorTypes").setParameter("entryId", entryId).getResultList();
+        return (List<String>)result.stream().map(descriptorLanguage -> descriptorLanguage.toString()).collect(Collectors.toList());
+    }
```

Comment:
For these two i want to ensure a few things:

1. It's made abundantly clear we're grabbing a single record from the database and the methods return something based on that single record
2. If it's somehow getting no records (a big problem), there would be a log error message or customwebexception or something
3. If it's somehow getting multiple records  (also a big problem), there would be a log message or customwebexception or something

Reply:
2 is addressed, right now it's working properly, the error is gone(not really sure why). So we can ignore 3, will use `getSingleResult` - 1